src/megachatapi_impl.cpp

```diff
@@ -5857,17 +6165,18 @@ void MegaChatCallHandler::onStateChange(uint8_t newState)
                 state = MegaChatCall::CALL_STATUS_IN_PROGRESS;
                 break;
             case rtcModule::ICall::kStateTerminating:
-                state = MegaChatCall::CALL_STATUS_TERMINATING;
+                state = MegaChatCall::CALL_STATUS_TERMINATING_USER_PARTICIPATION;
                 chatCall->setIsRinging(false);
                 chatCall->setTermCode(call->termCode());
                 chatCall->setFinalTimeStamp(time(NULL));
                 API_LOG_INFO("Terminating call. ChatId: %s, callid: %s, termCode: %s , isLocal: %d, duration: %d (s)",
-                             call->chat().chatId().toString().c_str(), call->id().toString().c_str(),
-                              rtcModule::termCodeToStr(call->termCode() & (~rtcModule::TermCode::kPeer)),
+                             karere::Id(chatCall->getChatid()).toString().c_str(),
+                             karere::Id(chatCall->getId()).toString().c_str(),
+                             rtcModule::termCodeToStr(call->termCode() & (~rtcModule::TermCode::kPeer)),
                              chatCall->isLocalTermCode(), chatCall->getDuration());
                 break;
             case rtcModule::ICall::kStateDestroyed:
-                state = MegaChatCall::CALL_STATUS_DESTROYED;
+                return;
```

Comment:
Should we use `MegaChatCall::CALL_STATUS_USER_NO_PRESENT` in this case?

Reply:
I haven't notified `MegaChatCall::CALL_STATUS_USER_NO_PRESENT` at `kStateDestroyed` to avoid notify twice the same state to apps. The call state flow is `in-progress`->`terminating`->`destroyed`. We notify `MegaChatCall::CALL_STATUS_USER_NO_PRESENT` at `kStateTerminating` then I think that it's not necessary notify at `kStateDestroyed` too